### PR TITLE
MTDSA-6965 Updated models for individual calculations (backend)

### PR DESCRIPTION
**Pull request by Taiwo-AB**

### CODE REVIEW

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import support.UnitSpec
+import v1.fixtures.getCalculation.endOfYearEstimate.detail.EoyEstimateUkSecuritiesFixtures._
+
+class EoyEstimateUkSecuritiesSpec extends UnitSpec {
+
+  "EoyEstimateUkSecurities" when {
+    "read from valid Json" should {
+      "return a JsSuccess" in {
+        eoyEstimateUkSecuritiesDesJson.validate[EoyEstimateUkSecurities] shouldBe a[JsSuccess[_]]
+      }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala:29]: This test step is redundant.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateStateBenefitsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import support.UnitSpec
+import v1.fixtures.getCalculation.endOfYearEstimate.detail.EoyEstimateStateBenefitsFixtures._
+
+class EoyEstimateStateBenefitsSpec extends UnitSpec {
+
+  "EoyEstimateStateBenefits" when {
+    "read from valid Json" should {
+      "return a JsSuccess" in {
+        eoyEstimateStateBenefitsDesJson.validate[EoyEstimateStateBenefits] shouldBe a[JsSuccess[_]]
+      }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateStateBenefitsSpec.scala:29]: This test step is redundant.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateOtherDividendsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import support.UnitSpec
+import v1.fixtures.getCalculation.endOfYearEstimate.detail.EoyEstimateOtherDividendsFixtures._
+
+class EoyEstimateOtherDividendsSpec extends UnitSpec {
+
+  "EoyEstimateOtherDividends" when {
+    "read from valid Json" should {
+      "return a JsSuccess" in {
+        eoyEstimateOtherDividendsDesJson.validate[EoyEstimateOtherDividends] shouldBe a[JsSuccess[_]]
+      }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateOtherDividendsSpec.scala:29]: This test step is redundant.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import support.UnitSpec
+import v1.fixtures.getCalculation.endOfYearEstimate.detail.EoyEstimateForeignPropertyFixtures._
+
+class EoyEstimateForeignPropertySpec extends UnitSpec {
+
+  "EoyEstimateForeignProperty" when {
+    "read from valid Json" should {
+      "return a JsSuccess" in {
+        eoyEstimateForeignPropertyDesJson.validate[EoyEstimateForeignProperty] shouldBe a[JsSuccess[_]]
+      }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala:29]: This test step is redundant.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      "with the expected EoyEstimateForeignProperty object" in {
+        eoyEstimateForeignPropertyDesJson.as[EoyEstimateForeignProperty] shouldBe eoyEstimateForeignPropertyResponse
+      }
+    }
+
+    "read from Json with missing optional fields" should {
+      "return the expected EoyEstimateForeignProperty object" in {
+        eoyEstimateForeignPropertyDesJsonMissingFields.as[EoyEstimateForeignProperty] shouldBe eoyEstimateForeignPropertyResponseFactory(finalised = None)
+      }
+    }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala:39]: I don't think this needs testing.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      "return the expected JsObject" in {
+        Json.toJson(eoyEstimateForeignPropertyResponse) shouldBe eoyEstimateForeignPropertyWrittenJson
+      }
+    }
+
+    "written to Json with missing optional fields" should {
+      "return the expected JsObject" in {
+        Json.toJson(eoyEstimateForeignPropertyResponseFactory(finalised = None)) shouldBe eoyEstimateForeignPropertyWrittenJsonMissingFields
+      }
+    }
```
Comment (BetaDraconis) [inline test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertySpec.scala:57]: I don't think this needs testing.

File: test/v1/fixtures/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertyFixtures.scala [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  val eoyEstimateForeignPropertyWrittenJsonObject: JsObject = Json.obj("foreignProperty" -> eoyEstimateForeignPropertyWrittenJson)
+
+  val eoyEstimateForeignPropertyInvalidJson: JsValue = Json.parse(s"""{
+      |  "taxableIncome" : $taxableIncome,
+      |  "finalised" : "notABoolean"
+      |}""".stripMargin)
+
+  def eoyEstimateForeignPropertyResponseFactory(taxableIncome: BigInt = taxableIncome,
+                                                finalised: Option[Boolean] = finalised): EoyEstimateForeignProperty =
+    EoyEstimateForeignProperty(taxableIncome, finalised)
```
Comment (BetaDraconis) [inline test/v1/fixtures/getCalculation/endOfYearEstimate/detail/EoyEstimateForeignPropertyFixtures.scala:64]: Not sure this is necessary for such a small case class.

**Review by BetaDraconis — CHANGES_REQUESTED** [review]
A couple of redundant test scenarios.

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala (hunk 1 above)
Comment (Taiwo-AB) [reply test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala:29]: All were done for consistency with what is in the code base

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateStateBenefitsSpec.scala (hunk 2 above)
Comment (Taiwo-AB) [reply test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateStateBenefitsSpec.scala:29]: Same here

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateOtherDividendsSpec.scala (hunk 3 above)
Comment (Taiwo-AB) [reply test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateOtherDividendsSpec.scala:29]: and the others also

File: test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala (hunk 1 above)
Comment (BetaDraconis) [reply test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala:29]: Good > consistent

Comment (Taiwo-AB) [reply test/v1/models/response/getCalculation/endOfYearEstimate/detail/EoyEstimateUkSecuritiesSpec.scala:29]: Yh but it always keeps changing. Will make the changes

**Review by BetaDraconis — APPROVED** [review]

**Pull request merged**